haigha/transports/socket_transport.py

```diff
@@ -29,23 +29,33 @@ def connect(self, (host, port), klass=socket.socket):
         Connect assuming a host and port tuple.
         '''
         self._host = "%s:%s" % (host, port)
-        self._sock = klass()
+        infos = socket.getaddrinfo(host, port, 0, 0, socket.IPPROTO_TCP)
+        if not infos:
+            raise socket.error("getaddrinfo returns an empty list")
+
+        # Sort addresses such that the new IPv6 address is first.
+        for info in reversed(sorted(infos, key=lambda info: len(info[-1]))):
+
+            sockaddr = info[-1]
+            try:
+                # Check if the IP is a valid IPv6 address. Fallback to IPv4 if not.
+                socket.inet_pton(socket.AF_INET6, sockaddr[0])
```

Comment:
@kevinconway, there is no need to determine the appropriate family, because getaddrinfo returns everything that's needed already, including the address family value to use with the corresponding address.